File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        int m = part.size();
        while(s.find(part)!=string::npos){ // loop runs till there is no 'part' substr left
            int i= s.find(part); // finds index of first occurred part substr
            s.erase(i,m);
        }
        return s;

    }
};

File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        int m = part.length();

        while (s.contains(part)) {
            int sIdx = s.indexOf(part);
            s = s.substring(0, sIdx) + s.substring(sIdx + m);
        }

        return s;
    }
}
